mixed-content/generic/common.js

```diff
@@ -0,0 +1,238 @@
+function normalizePort(targetPort) {
+  var defaultPorts = [80, 443];
+  var isDefaultPortForProtocol = (defaultPorts.indexOf(targetPort) >= 0);
+
+  return (targetPort == "" || isDefaultPortForProtocol) ?
+          "" : ":" + targetPort;
+}
+
+function setAttributes(el, attrs) {
+  attrs = attrs || {}
+  for (var attr in attrs)
+    el[attr] = attrs[attr];
+}
+
+function createElement(tagName, attrs, parent) {
+  var el = document.createElement(tagName);
+  setAttributes(el, attrs);
+  if (parent && parent.appendChild)
+    parent.appendChild(el);
+
+  return el;
+}
+
+function createHelperIframe(name) {
+  return createElement("iframe", {"name": name}, document.body);
+}
+
+function queryIframe(url) {
+  createElement("iframe", {"src": url}, document.body);
+}
+
+function queryImage(url) {
+  createElement("img", {"src": url}, document.body)
+}
+
+function queryXhr(url) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url, true);
+  xhr.responseType = "json";
+  xhr.send();
```

Comment:
If we look at all types of resources we cover, I think there isn't really a nice way to get callbacks for each (e.g. did the request go through or not). E.g. if we load an image via CSS.

 I'll see how much we can cover and possibly add additional checks. In any case, some onload events and other XHR events should be least of the problems.

The test itself is reusing server-side state to confirm if the requests went through or not. That's the overall idea, however the timeout for the assertion after the request can be a potential cause of flakiness.